### DISCUSSION TITLE
issue: useFieldArray defaultValues not populating form

### DISCUSSION THREAD

**Discussion by pwharff** [react-hook-form/react-hook-form#7173] [Q&A] [answered]
### Version Number

7.15.3

### Codesandbox/Expo snack

https://stackblitz.com/edit/react-hook-form-7-dynamic-form-example-nidvxr?file=App.jsx

### Steps to reproduce

1. Open the code sandbox
2. See `Full name` default value of `Joe Schmoe` not populated

When using `useFieldArray` and `defaultValues`, no values are populated in the form:

```
const { control, errors, handleSubmit } = useForm({
    defaultValues: {
      accounts: [{ fullName: 'Joe Schmoe' }],
    },
  });
```

See stack blitz example.

### Expected behaviour

To populate the form with default values by providing object to `defaultValues` argument of `useForm`

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
can we provide a reasonable example for me to look into instead of dumping a partial project here? here is a template you can fork and use: https://codesandbox.io/s/react-hook-form-usefieldarray-ssugn

**Reply by pwharff** [reply]
I can sure do that, but I did provide a fully working example with the Stack Blitz link. Would you rather I convert it to using the template you provided? If so, I'd be glad to do so.

**Reply by bluebill1049** [reply]
can you reproduce the issue in the above example? without all those external dependencies.

**Reply by pwharff** [reply]
Sorry for taking so long. Ok, I implemented using your template: 

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-x8dwy?file=/src/index.js

I think my issue is more to do with how I setup the Controller for `TextField` or `NumberField` inputs, as default values are still not populated and appended values aren't working either. 

Any help is greatly appreciated. Thanks.

**Reply by pwharff** [reply]
Interestingly enough, the form is submitted with the default values and not those from the input:

<img src="https://user-images.githubusercontent.com/31776351/144123881-031c0c32-af8e-4209-a603-d6bd0837743f.png" width="500">

**Comment by bluebill1049** [comment]
please refer to the doc: https://react-hook-form.com/api/usecontroller/controller

<img width="964" alt="Screen Shot 2021-12-01 at 9 11 29 am" src="https://user-images.githubusercontent.com/10513364/144136347-8345a794-a67e-4e95-b9cd-92d9ea8ffb15.png">

`render={({ field: {value, onChange} }) => {`

**Reply by bluebill1049** [reply]
<img width="1031" alt="Screen Shot 2021-12-01 at 9 12 19 am" src="https://user-images.githubusercontent.com/10513364/144136468-03f02f1e-f42a-4f93-92d6-9b6825c9a0cd.png">

the document example has such example as well.

**Comment by pwharff** [comment, marked as answer]
This was the key:

`render={({ field: {value, onChange} }) => {`

Many, many thanks!!! My version of `react-hook-form` didn't have `field`, but I updated and viola! All is good. 

Thanks again. Love this project, I'll definitely be donating.
